src/Microsoft.AspNetCore.Server.Testing/Deployers/SelfHostDeployer.cs

```diff
@@ -53,7 +53,8 @@ private CancellationToken StartSelfHost(Uri uri)
             string executableArgs = string.Empty;
             if (DeploymentParameters.PublishApplicationBeforeDeployment)
             {
-                executableName = Path.Combine(DeploymentParameters.PublishedApplicationRootPath, new DirectoryInfo(DeploymentParameters.ApplicationPath).Name);
+                var executableExtension = DeploymentParameters.RuntimeFlavor == RuntimeFlavor.Clr ? ".exe" : "";
```

Comment:
Should we even check for the OS platform as even in case of Mono on Linux/Mac, we do not want this extension